cwms-http-client/src/main/java/mil/army/usace/hec/cwms/http/client/HttpRequestBuilderImpl.java

```diff
@@ -42,10 +43,14 @@
     private final Map<String, String> queryParameters = new HashMap<>();
     private final Map<String, String> queryHeaders = new HashMap<>();
 
-    public HttpRequestBuilderImpl(ApiConnectionInfo apiConnectionInfo, String endpoint) {
+    public HttpRequestBuilderImpl(ApiConnectionInfo apiConnectionInfo, String endpoint) throws ServerNotFoundException {
         Objects.requireNonNull(apiConnectionInfo, "API connection info must be defined");
         Objects.requireNonNull(apiConnectionInfo.getApiRoot(), "API root must be defined");
-        this.httpUrl = HttpUrl.parse(apiConnectionInfo.getApiRoot());
+        HttpUrl url = HttpUrl.parse(apiConnectionInfo.getApiRoot());
```

Comment:
do you know if this supports ipv6?

Reply:
It does, I added a unit test to demonstrate. It also cleans it up.